src/Aspire.Hosting/ApplicationModel/EndpointAnnotation.cs

```diff
@@ -104,6 +105,11 @@ public int? TargetPort
     /// </summary>
     public string UriScheme { get; set; }
 
+    /// <summary>
+    /// This is the address the resource is listening on. By default it is localhost.
+    /// </summary>
+    public string TargetHostAddress { get; set; } = "localhost";
```

Comment:
New public API. I could keep internal, but it felt better to expose this in WithEndpoint callbacks. This is to address as targetport is to port.

Reply:
Would `TargetAddressHost` be a more correct name? Or just `TargetHost`? If we go with the `BindingAddress` terminology, 'address' represents everything, and the host is just the one part that's the host name or IP address.

Reply:
I like TargetHost.